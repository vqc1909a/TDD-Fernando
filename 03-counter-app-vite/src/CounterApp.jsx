import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    
    const [ counter, setCounter ] = useState( value );

    const addHandler = () => { 
        // console.log(event)
        setCounter( counter + 1 );
        // setCounter( (c) => c + 1 )
    }

    const substractHandler = () => setCounter( counter - 1 );
    const resetHandler = () => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={ addHandler }> +1 </button>
            <button onClick={ substractHandler }> -1 </button>
            <button aria-label="btn-reset" onClick={ resetHandler }> Reset </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}


