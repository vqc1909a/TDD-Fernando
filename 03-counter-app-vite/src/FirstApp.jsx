import PropTypes from 'prop-types';


const getResult = () => (
  <>
    <h1>Subtitle</h1>
  </>
)

export const FirstApp = ({ title, subTitle, name, number }) => {
  const newMessage = {
    message: "Hola Mundo",
    title: "Fernando"
  }

  return (
    //If the father tag makes no sense in the DOM tree, then just put etiquetes empty as the next example below
    <>
      <h1 data-testid="test-title">{ title }</h1>
      {/* <pre>
        <code>{ JSON.stringify(newMessage, null, 2) }</code>
      </pre>    */}
      {/* <h2>{getResult()}</h2> */}
      <h3>Number: {number + 2}</h3>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}

//These values default are assigned to properties before its be validated with propTypes
FirstApp.defaultProps = {
  name: 'No hay nombre',
  number: 0,
}

//if you are working with vite you need to install the package prop-types, otherwise if you are working with create-react-app, is not necessary install prop-types, as that it comes installed by default 
FirstApp.propTypes = {
  name: PropTypes.string,
  number: PropTypes.number,
  subTitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}

