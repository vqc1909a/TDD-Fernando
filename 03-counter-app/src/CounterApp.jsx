import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    const [ counter, setCounter ] = useState( value );

    const handleAdd = (event, name) => { 
        console.log({event, name})
        setCounter( counter + 1 );
        // setCounter( (c) => c + 1 )
    };

    const handleSubstract = (event) => counter === 0  ? setCounter(0) : setCounter( counter - 1 );

    const handleReset = (event) => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={ (e) => handleAdd(e, "Goku") }> +1 </button>
            <button onClick={ handleSubstract }> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
        </>
    );
}   

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}


