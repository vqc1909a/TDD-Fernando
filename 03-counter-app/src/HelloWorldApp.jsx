export const HelloWorldApp = () => {
  return (
    <h1>Hello World App Vite</h1>
  );
}
