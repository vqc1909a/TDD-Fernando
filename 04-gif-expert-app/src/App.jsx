import { useState } from 'react'
import './App.css'
import { AddCategory, GifGridCategory } from "./components/Gif";
import { nanoid } from 'nanoid'

const DEFAULT_CATEGORIES = [
  {
    name: 'One Puch',
    id: nanoid()
  }
]
function App() {

  const [categories, setCategories] = useState(DEFAULT_CATEGORIES)

  const handleAddCategory = (category) => {
    const newCategory = {
      name: category,
      id: nanoid()
    }
    setCategories(categories => [newCategory, ...categories]);
  }

  return (
    <div>
     <h1>GifExpertApp</h1> 
     <AddCategory handleAddCategory={handleAddCategory}></AddCategory>
     {/* React recommends not use the parameter index for the keys, because it could be a problem when the list changes. The best way is use a unique key, like the id of the object brought out of the npm package or the db. */}
     <ol>
      {
        categories.map(({ id, name}) => <GifGridCategory key={id} category={name}></GifGridCategory>)
      }
     </ol>
    </div>
  )
}

export default App
