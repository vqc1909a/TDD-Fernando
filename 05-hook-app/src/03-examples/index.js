export * from './MultipleCustomHooks';
export * from './LoadingQuote';
export * from './Quote';
export * from './ErrorQuote';

