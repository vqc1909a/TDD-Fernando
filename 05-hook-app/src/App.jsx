
function App() {

  return (
    <>
      <h1>HooksApp</h1>
    </>
  )
}

export default App
