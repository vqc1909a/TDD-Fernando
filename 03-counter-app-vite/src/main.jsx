import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import './styles.css';



ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        {/* if you not assign a value to some property, the property's value by default is true  */}
        {/* <FirstApp title="Hola, Soy Vegeta" subTitle="Hola, Soy el subtitulo" /> */}
        <CounterApp value={ 20 } />
        {/* <HelloWorldApp /> */}

    </React.StrictMode>
);


