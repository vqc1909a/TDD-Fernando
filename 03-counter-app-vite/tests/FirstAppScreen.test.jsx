import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp /> with screen', () => {

    const title = "Hola, soy Goku";
    const subTitle =  "Hola, Soy el subtitulo";

    test('should make to match to the snapshot', () => { 
        const {container} = render(<FirstApp title={title} subTitle={subTitle} />)
        expect(container).toMatchSnapshot();
    })

    test('should show the message "Hola, Soy Goku"', () => { 
        
        //screen is the object that we be rendering in one each test with the sentence render, of way automatic 
        render(<FirstApp title={title} subTitle={subTitle} />)
        screen.debug(); /* With this sentence we look the component rendered, i same the html generated, in a first instance it's be the tag <body />, but if before we rendering a component, then the screen will be the componente rendering */
        expect(screen.getByText(title)).toBeTruthy ();
    })

    test('should show title in one h1', () => {
        render(<FirstApp title={title} subTitle={subTitle} />)
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
        expect(screen.getByRole('heading', { level: 1 }).textContent).toContain(title);
    })

    test('should show subTitles that have be send in props', () => {
        render(<FirstApp title={title} subTitle={subTitle} />)
        expect(screen.getAllByText(subTitle)).toBeTruthy();
        expect(screen.getAllByText(subTitle).length).toBe(2);
        expect(screen.getAllByText(subTitle)[0].textContent).toContain(subTitle);
        expect(screen.getAllByText(subTitle)[0].innerHTML).toContain(subTitle);


    })
});