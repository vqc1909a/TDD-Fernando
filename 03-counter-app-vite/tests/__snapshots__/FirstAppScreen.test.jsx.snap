// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> with screen should make to match to the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <h2>
    Hola, Soy el subtitulo
  </h2>
  <h2>
    Hola, Soy el subtitulo
  </h2>
  <p>
    Number: 
    2
  </p>
  <p>
    No hay nombre
  </p>
</div>
`;
